mlir/lib/Gradient/Utils/DestinationPassingStyle.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+
+#include "Gradient/Utils/DestinationPassingStyle.h"
+
+using namespace mlir;
+
```

Comment:
It might be worth to have a small doc here (distinction between memref and non memref)

Reply:
Right, there is already a doc on the forward declaration of this function in the corresponding header. Would you like me to move it here? Or are you proposing to document something separate here?

Reply:
I missed that sorry, all good on my side then :+1: